cmd/cluster/support/delete.go

```diff
@@ -123,11 +122,11 @@ func (o *deleteOptions) run() error {
 }
 
 // createDeleteRequest sets the delete API and returns a request
-func createDeleteRequest(ocmClient *sdk.Connection, cluster *v1.Cluster, reasonID string) (request *sdk.Request, err error) {
+func createDeleteRequest(client SDKConnectionClient, clusterID string, reasonID string) (request *sdk.Request, err error) {
```

Comment:
ocmClient seemed more specific, why did you change it? 

Also does this work on a real cluster? I think deleting by external ID doesn't work, but I see that's what your test is running. Can this be changed back so we aren't altering the behavior we're testing as well?

Reply:
You cannot delete by external ID in OCM, only by internal ID. So I think leaving this as `cluster *v1.Cluster` is better here because we KNOW a v1.Cluster has the internal ID as part of it, where as with a string, there's no guarantee about the data passed in.